### PR TITLE
User profile class

### PR DESCRIPTION
**Pull request by lmartin73**

### CODE REVIEW

File: src/app/_be/auth/useracct.js [hunk 1]
```diff
@@ -0,0 +1,199 @@
+import * as firebase from 'firebase'
+import {FIRDATAprofilepaths} from '../../config/firebaseConfig'
```
Comment (lmartin73) [inline src/app/_be/auth/useracct.js]: Do not need to import firebase data paths

File: src/app/_be/auth/useracct.js [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+
+export class ACCTprfl {
```
Comment (jhubbard2017) [inline src/app/_be/auth/useracct.js]: Modify class name to a more readable form.

File: src/app/_be/auth/useracct.js [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+        this.profileDataPath = `users/${this.uid}`;
+        this.profileStoragePath = `users/${this.uid}/`
+        if(uid != ""){
+            this.downloadProfile();
+        }
+    }
+    /* Create a user profile in our database for a user who has used an email and password to sign up */
+    createProfileViaEmail(firstname, lastname, email, uid){
+        this.uid = uid;
+        this.profileDataPath = `users/${this.uid}`;
```
Comment (lmartin73) [inline src/app/_be/auth/useracct.js:29]: you need to set the storage data path.

File: src/app/_be/auth/useracct.js [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+    };
+
+    /* Create a user profile in our database for a user who has used Facebook or Google to sign up */
+    createProfileViaProvider(user){
+        this.uid = user.uid;
+        this.profileDataPath = `users/${this.uid}`;
+        this.profileStoragePath = `users/${this.uid}/`
+        var displayName = user.displayName;
+        var name = displayName.split(" ");
+        this.setName(name[0],name[1]);
```
Comment (jhubbard2017) [inline src/app/_be/auth/useracct.js]: Include logic to handle a middle name being present in the display name.

File: src/app/_be/auth/useracct.js [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+
+    setEmail(email=""){
+        firebase.database().ref(this.profileDataPath).update({
+            email: email
+        });
+    }
+
+    setProfileImg(file=null){
+        var imgStorageRef = firebase.storage().ref();
+        this.profilePhoto = imgStorageRef.child(`${this.profileStoragePath}\profileimg`);
```
Comment (lmartin73) [inline src/app/_be/auth/useracct.js]: profilePhoto should only have scope to this routine not the class.

**Review by jhubbard2017 — CHANGES_REQUESTED** [review]

File: src/app/_be/auth/useracct.js (hunk 1 above)
Comment (lmartin73) [reply src/app/_be/auth/useracct.js]: Done

File: src/app/_be/auth/useracct.js (hunk 3 above)
Comment (lmartin73) [reply src/app/_be/auth/useracct.js:29]: Done

File: src/app/_be/auth/useracct.js (hunk 5 above)
Comment (lmartin73) [reply src/app/_be/auth/useracct.js]: Done

File: src/app/_be/auth/useracct.js (hunk 2 above)
Comment (lmartin73) [reply src/app/_be/auth/useracct.js]: Done

File: src/app/_be/auth/useracct.js (hunk 4 above)
Comment (lmartin73) [reply src/app/_be/auth/useracct.js]: Done

**Review by jhubbard2017 — APPROVED** [review]

**Pull request merged**